core/include/utils/math_util.h

```diff
@@ -15,4 +16,38 @@ double clamp(double value, double low, double high);
 * 
 * returns the sign +/-1 of x. 0 if x is 0
 **/
-double sign(double x);
\ No newline at end of file
+double sign(double x);
+
+
+/*
+Calculates the variance of  a set of numbers (needed for linear regression)
+https://en.wikipedia.org/wiki/Variance
+@param values   the values for which the covariance is taken
+@param mean     the average of values
+*/
+double variance(std::vector<double> const &values, double mean);
+
+
+/*
+Calculates the average of a vector of doubles
+@param values   the list of values for which the average is taken
+*/
+double mean(std::vector<double> const &values);
+
+/*
+Calculates the covariance of a set of points (needed for linear regression)
+(refactor to accept to sets of values not a set of points)
```

Comment:
Also just a note to myself I forgot to get rid of. I originally wrote it to use one vector of doubles for x and a vector of doubles for y rather than the vector of pairs of x and y.
```suggestion
```